packages/create-novu-app/templates/app-react-email/ts/app/novu/emails/novu-onboarding-email.tsx

```diff
@@ -0,0 +1,194 @@
+import {
+  Body,
+  Button,
+  Column,
+  Container,
+  Head,
+  Html,
+  Img,
+  Preview,
+  Row,
+  Section,
+  Text,
+  Hr,
+  Tailwind,
+  render,
+} from "@react-email/components";
+import * as React from "react";
+import { ControlSchema, EmailComponent, PayloadSchema } from "../workflows";
+
+interface NovuWelcomeEmailProps {
+  steps?: {
+    id: number;
+    Description: React.ReactNode;
+  }[];
+  welcomeHeaderText?: string;
+  belowHeaderText?: string;
+  colorChange?: string;
+  components?: EmailComponent[];
+  userImage: string;
+  teamImage: string;
+  arrowImage: string;
+}
```

Comment:
Can we replace this with a `z.infer`red union of the control+payload schemas combined? This will be a super powerful story demonstrating end to end type safety of a Novu integration.

We shouldn't need to spell out the full props here, instead let them be decided by the schemas. Wdyt?

Reply:
Great idea, done!